### ISSUE TITLE
[MPOM-244] Deploy SHA-512 for source-release to Remote Repository

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-apache-parent#456]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** opened **[MPOM-244](https://issues.apache.org/jira/browse/MPOM-244?redirect=false)** and commented

As now the ASF staging repository supports SHA2 hashes (https://issues.apache.org/jira/browse/INFRA-14923) they should be generated for
- the ASF source release artifact (as recommended in https://infra.apache.org/release-distribution.html#sigs-and-sums)
- deployed to the Staging repository (i.e. attached to build as well)

---

**Affects:** ASF-23

**Issue Links:**
- [MRESOLVER-56](https://issues.apache.org/jira/browse/MRESOLVER-56) Support SHA-256 and SHA-512 as checksums
 (_**"is blocked by"**_)
- [MINSTALL-138](https://issues.apache.org/jira/browse/MINSTALL-138) option to calculate more checksum such sha-256 sha-512

- [SLING-7534](https://issues.apache.org/jira/browse/SLING-7534) Release policy - stop providing MD5 and start providing SHA-512 checksums

- [MPOM-282](https://issues.apache.org/jira/browse/MPOM-282) Create correct SHA512 content

- [MPOM-205](https://issues.apache.org/jira/browse/MPOM-205) create SHA-512 checksum for source-release archive(s) in target/checkout/target/ during release
 (_**"depends upon"**_)

**Remote Links:**
- [GitHub Pull Request #40
](https://github.com/apache/maven-apache-parent/pull/40)

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

This has to be configured in the release plugin to pass properties to the forked Maven instance for Resolver to pick up.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

That sounds good. Is it possible to update to a newer resolver with the approach from MNG-5771 or does this require waiting for Maven 4?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

No, you need to replace Resolver completely,  but this isn't an issue with 3.6.3.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

Manual steps fiddling with your Maven installation are IMHO too much effort. I guess this can only be fixed once Maven 4 is released and enforced...

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

If you really need it, you will do it.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

Instead of configuring this in the Maven Resolver, we should generate the SHA512 checksums as additional artifacts with http://checksum-maven-plugin.nicoulaj.net/artifacts-mojo.html.  That way the Staging Repository contains all necessary artifacts for an ASF release in dist. The only thing which currently blocks is a release containing https://github.com/nicoulaj/checksum-maven-plugin/pull/87 and https://github.com/nicoulaj/checksum-maven-plugin/pull/101.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

It is important to understand that the checksums generated by Resolver are private to Revolver transport only and are not artifacts. They are out of your reach. The checksum plugin is not related to that.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

In the past the Maven generated SHA1 checksum has been used by a lot of ASF projects as checksum which was pushed to dist once finalizing a release. Now that ASF has the requirement for SHA512 in dist this is on longer possible, but as I said, generating the SHA512 checksums with checksum-maven-plugin, attach them as 2nd artifacts and deploy them to the staging repo allows to use the same approach.

That allows also to easily verify a release from just the contents of the stage repository....

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

You are confusing dist with Central. These are two completely different things. We do also create source release with checksums for dist. Look at our parent POM.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

I want everything which is pushed to dist in the staging repo. This was like that in the past when only SHA1 was required. The ASF parent pom currently creates SHA512 but does not deploy those to the staging repo forcing the one finalizing the release regenerating it.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

So you copy everything from Nexus to dist?

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o) We at Apache Sling do that, but more important is that no additional files are added to dist which are not in the Staging Repo, as the vote happens on the contents in the stage repo and that should contain everything about to be released (to Maven Central and/or dist). Also the source-release artifacts are always attached (https://github.com/apache/maven-apache-parent/blob/e872a8057d93749d6b3c8ed2d7d3b064076085ab/pom.xml#L382).

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

PR merged in https://github.com/apache/maven-apache-parent/commit/b08ef161f1847a17b36b456f68bfa784770b6a58

thanks a lot [Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin) for the hard work on finding the best compromise, which was not easy

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

The plugin fails when using Maven 3.1.1 and calling `mvn verify -Papache-release`

The minimalMavenBuildVersion is 3.1.1, however the plugin net.nicoulaj.maven.plugins:checksum-maven-plugin:1.10 requires Maven version 3.2.5.
When using Maven 3.1.1 you cannot skip this plugin.
Either the minimum should be lifted to 3.2.5 or (which I prefer) the checksum-maven-plugin should be moved to a profile as done for the maven-artifact-plugin:

```xml
<profile>
  <id>pgpverify</id>
  <activation>
    <property>
      <name>pgpverify.skip</name>
      <value>!true</value>
    </property>
  </activation>
  <build>
    <plugins>
      <plugin>
        <groupId>org.simplify4u.plugins</groupId>
        <artifactId>pgpverify-maven-plugin</artifactId>
        <executions>
          <execution>
            <goals>
              <goal>check</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>
</profile>
```

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

2014-12-20 	3.2.5
Let's lift in a separate PR!

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

PR for checksum-maven-plugin done https://github.com/nicoulaj/checksum-maven-plugin/pull/118

**Comment by jira-importer** [comment]
**[Julien Nicoulaud](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nicoulaj)** commented

Hi guys, I just released version 3.11 of checksum-maven-plugin that restores compatibility with Maven 3.1.1.

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

plugin upgraded, thanks a lot [Julien Nicoulaud](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nicoulaj)
